initializr/src/main/groovy/io/spring/initializr/config/InitializrMetricsExporterAutoConfiguration.groovy

```diff
@@ -0,0 +1,165 @@
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass
+import org.springframework.boot.autoconfigure.condition.OnBeanCondition
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition
+import org.springframework.boot.autoconfigure.redis.RedisAutoConfiguration
+import org.springframework.boot.autoconfigure.redis.RedisProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.cloud.CloudException
+import org.springframework.cloud.CloudFactory
+import org.springframework.cloud.service.common.RedisServiceInfo
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.ConditionContext
+import org.springframework.context.annotation.Conditional
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Primary
+import org.springframework.core.type.AnnotatedTypeMetadata
+import org.springframework.core.type.StandardAnnotationMetadata
+import org.springframework.data.redis.connection.RedisConnectionFactory
+import org.springframework.data.redis.connection.jedis.JedisConnection
+import org.springframework.data.redis.core.RedisOperations
+import org.springframework.scheduling.annotation.EnableScheduling
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.util.ObjectUtils
+
+import redis.clients.jedis.Jedis
+
+/**
+ * @author Dave Syer
+ *
+ */
+@Configuration
+@Conditional(RedisDetected)
+@EnableScheduling
+@EnableConfigurationProperties(MetricsProperties)
+@AutoConfigureAfter(RedisAutoConfiguration)
+class InitializrMetricsExporterAutoConfiguration {
+
+	@Autowired
+	RedisConnectionFactory connectionFactory
+
+	@Autowired
+	MetricsProperties metrics
+
+	@Autowired
+	ApplicationContext context
+
+	@Bean
+	MetricWriter writer() {
+		new RedisMetricRepository(connectionFactory,
+				metrics.prefix + metrics.getId(context.getId()) + '.'
+				+ ObjectUtils.getIdentityHexString(context) + '.',
+				metrics.key)
+	}
+
+	@Bean
+	@Primary
+	MetricRepository reader() {
+		new InMemoryMetricRepository()
+	}
+
+	@Bean
+	Exporter exporter(InMemoryMetricRepository reader) {
+		new MetricCopyExporter(reader, writer()) {
+					@Override
+					@Scheduled(fixedRateString = '${initializr.metrics.rateMillis:5000}')
+					void export() {
+						super.export()
+					}
+				}
+	}
+
+	@Bean
+	@Primary
+	@Conditional(CloudRedisDetected)
+	RedisProperties redisProperties() throws UnknownHostException {
+		RedisProperties properties = new RedisProperties()
+		CloudFactory cloud = new CloudFactory()
+		RedisServiceInfo serviceInfo = (RedisServiceInfo) cloud.getCloud()
+				.getServiceInfo('redis')
+		properties.setHost(serviceInfo.getHost())
```

Comment:
so the service on PCF should be named redis? Isn't that a bit weird to have something cloud foundry specific instead of just detecting plain redis access to the application?